yarn-project/aztec-rpc/src/contract_data_oracle/index.ts

```diff
@@ -43,6 +44,34 @@ export class ContractDataOracle {
     return tree.getFunctionAbi(functionSelector);
   }
 
+  /**
+   * Retrieves the debug metadata of a specified function within a given contract.
+   * The function is identified by its selector, which is a unique code generated from the function's signature.
+   * Returns undefined if the debug metadata for the given function is not found.
+   *
+   * @param contractAddress - The AztecAddress representing the contract containing the function.
+   * @param functionSelector - A Buffer containing the unique selector code for the desired function.
+   * @returns The corresponding function's ABI as an object.
+   */
+  public async getFunctionDebugMetadata(
```

Comment:
Maybe this can be moved to foundation:
`getFunctionDebugMetadata(abi: ContractAbi, functionName: string): FunctionDebugMetadata;`

Although debug in only used in simulator. But if this util lives with `ContractAbi`, it will be easier for people to see how data in `ContractAbi.debug` should be used, how it gets transformed into `FunctionDebugMetadata`. And the utils can be used here and in the test.

Reply:
Makes sense!